Answer the questions or complete the tasks outlined in bold below, use the specific method described if applicable.

** What is 7 to the power of 4?**

In [18]:
s = lambda num:num**4

In [19]:
s(7)

2401

In [58]:
7**4

2401

2401
** Split this string:**

s = "Hi there Sam!"
**into a list. **

In [20]:
s = "Hi there Sam!"

In [21]:
s.split()

['Hi', 'there', 'Sam!']

** Given the variables:**

planet = "Earth"
diameter = 12742
** Use .format() to print the following string: **

The diameter of Earth is 12742 kilometers.

In [24]:
planet = 'earth'
diameter = 12742

In [27]:
print('The diameter of {} is {} kilometers'.format(planet,diameter))

The diameter of earth is 12742 kilometers


** Given this nested list, use indexing to grab the word "hello" **

In [28]:
lst = [1,2,[3,4],[5,[100,200,['hello']],23,11],1,7]

In [57]:
lst[3][1][2][0]

'hello'

** Given this nested dictionary grab the word "hello". Be prepared, this will be annoying/tricky **

In [37]:
d = {'k1':[1,2,3,{'tricky':['oh','man','inception',{'target':[1,2,3,'hello']}]}]}

In [52]:
d['k1'][3]['tricky'][3]['target'][3]

'hello'

** What is the main difference between a tuple and a list? **

In [ ]:
# Tuple is immutable

** Create a function that grabs the email website domain from a string in the form: **

user@domain.com
So for example, passing "user@domain.com" would return: domain.com

In [64]:
def domainGet(email):
    return email.split('@')[1]

In [65]:
domainGet('user@domain.com')

'domain.com'

** Create a basic function that returns True if the word 'dog' is contained in the input string. Don't worry about edge cases like a punctuation being attached to the word dog, but do account for capitalization. **

In [66]:
s = "Is there dog here!"

In [78]:
def findDog(s):
    return  'dog' in s.lower().split()

In [79]:
findDog(s)

True

** Create a function that counts the number of times the word "dog" occurs in a string. Again ignore edge cases. **

In [80]:
st='This dog runs faster than the other dog dude!'

In [90]:
def countDog(st):
    count = 0
    for word in st.lower().split():
        if (word == 'dog'):
            count += 1
    return count

In [91]:
countDog(st)

2

** Use lambda expressions and the filter() function to filter out words from a list that don't start with the letter 's'. For example:**

seq = ['soup','dog','salad','cat','great']
should be filtered down to:

['soup','salad']


In [92]:
seq = ['soup','dog','salad','cat','great']

In [97]:
list (filter(lambda word:word[0]=='s',seq))

['soup', 'salad']


**You are driving a little too fast, and a police officer stops you. Write a function to return one of 3 possible results: "No ticket", "Small ticket", or "Big Ticket". If your speed is 60 or less, the result is "No Ticket". If speed is between 61 and 80 inclusive, the result is "Small Ticket". If speed is 81 or more, the result is "Big Ticket". Unless it is your birthday (encoded as a boolean value in the parameters of the function) -- on your birthday, your speed can be 5 higher in all cases. **

In [102]:
def caught_speeding(speed, is_birthday):
    if is_birthday:
        speeding = speed - 5
    else:
        speeding = speed
    
    if(speeding > 80):
         return 'big ticket'
    elif(speeding> 60 ):
         return 'small ticket'
    else :
         return 'no ticket'

In [104]:
caught_speeding(81,True)

'small ticket'

In [105]:
caught_speeding(80,False)

'small ticket'